src/Raven.Server/Documents/Sharding/ShardedDocumentDatabase.cs

```diff
@@ -215,10 +224,32 @@ public async Task DeleteBucketAsync(int bucket, long migrationIndex, long confir
         // to be applied in all orchestrator nodes
         await WaitForOrchestratorConfirmationAsync(confirmationIndex);
 
+        var delay = TimeSpan.FromSeconds(1);
         while (true)
         {
             var cmd = new DeleteBucketCommand(this, bucket, uptoChangeVector);
-            await TxMerger.Enqueue(cmd);
+            try
+            {
+                await TxMerger.Enqueue(cmd);
+            }
+            catch (Exception exception)
+            {
+                // if an error occurs during the execution of DeleteBucketCommand,
+                // we need to handle it gracefully to avoid rapid retry attempts
+
+                RaiseNotificationOnDeleteBucketFailure(bucket, exception);
+
+                if (delay >= TimeSpan.FromMinutes(5))
+                {
+                    // if the delay exceeds the maximum timeout, throw an exception and break the loop
+                    await Task.FromException(exception);
+                    return;
+                }
+
+                await Task.Delay(delay);
```

Comment:
this can be up to 5 min, right? so let's wire this with the database shutdown token

Reply:
done